marklogic-data-hub/src/test/java/com/marklogic/hub/dhs/DeployUserAmpsTest.java

```diff
@@ -0,0 +1,185 @@
+        DocumentManager adminDocManager = adminHubClient.getStagingClient().newDocumentManager();
+        writeSet = adminDocManager.newWriteSet();
+        writeSet.add("/pii/test.json", permissions, handle);
+        adminDocManager.write(writeSet);
+
+        ServerEvaluationCall piiData = operatorHubClient.getStagingClient().newServerEval().javascript("const ampMod = require('/ampTest/testModule.sjs');\n" +
+            "ampMod.getPiiData();");
+
+        //query returns null in the absence of the amp
+        assertFalse(piiData.eval().hasNext());
+
+        ResourcesFragment amps;
+        try{
+            //POST request
+            writeAmpFileToProjectAndDeploy(HubConfig.DEFAULT_MODULES_DB_NAME, "pii-reader");
+            amps = new AmpManager(runAsDataHubSecurityAdmin().getManageClient()).getAsXml();
+
+            assertTrue(amps.resourceExists("getPiiData"));
+            JacksonHandle jacksonHandle = new JacksonHandle();
+            piiData.eval().next().get(jacksonHandle);
+            assertEquals("123-456-7890", jacksonHandle.get().get("ssn").asText(), "data-hub-operator should be " +
+                "able to read the document since the amp has been created");
+
+            //PUT request - shouldn't fail
+            writeAmpFileToProjectAndDeploy(HubConfig.DEFAULT_MODULES_DB_NAME,"data-hub-developer", "pii-reader");
+        }
+        finally {
+            runAsDataHubSecurityAdmin().getManageClient().delete("/manage/v2/amps/getPiiData?namespace=&document-uri=/ampTest/testModule.sjs&modules-database=data-hub-MODULES");
+            amps = new AmpManager(runAsDataHubSecurityAdmin().getManageClient()).getAsXml();
+            assertFalse(amps.resourceExists("getPiiData"));
+        }
+    }
+
+    @Test
+    void createAmpWithDelegatedUser(){
+        //create custom role having "data-hub-amp-manager" as "data-hub-security-admin"
+        ObjectNode roleNode = mapper.createObjectNode();
+        roleNode.put("role-name", "delegated-amp-manager");
+        roleNode.set("role", mapper.createArrayNode().add("data-hub-amp-manager"));
+        writePayLoadToFileAndDeploy(getHubConfig().getUserSecurityDir().resolve("roles"), "delegatedRole.json", roleNode);
+
+        //create user from that role
+        ObjectNode userNode = mapper.createObjectNode();
+        userNode.put("user-name", "test-delegated-amp-manager");
+        userNode.put("password", "password");
+        userNode.set("role", mapper.createArrayNode().add("delegated-amp-manager"));
+        new UserManager(runAsAdmin().getManageClient()).save(userNode.toString());
+
+        //deploy amps using that user
+        ManageClient delegatedManageClient = new ManageClient();
+        delegatedManageClient.setManageConfig(new ManageConfig(host, 8002, "test-delegated-amp-manager", "password"));
+        try{
+            delegatedManageClient.postJson("/manage/v2/amps", getAmpNode("data-hub-MODULES", "data-hub-operator").toString());
```

Comment:
We want to make sure that this amp works too, so refactor the code in the test above for verifying that the amp works into a separate private method, and then call that method in both of these tests.